### PR TITLE
ssda903 schedule sensor

### PR DESCRIPTION
**Pull request by patrick-troy**
Add all of the necessary sensors and schedules (excluding the current WIP external dataset). This includes:
- clean_schedule
- reports_schedule
- move_current_sensor
- concatenate_sensor
- ssda903_fix_episodes_sensor
- move_error_reports_sensor
- move_current_and_concat_sensor
- sufficiency_sensor

### CODE REVIEW

**Review by MichaelHanksSF — APPROVED** [review]
This all seems to work just fine.

I have one question:
In common.checks.check_dataset(), why does this seem to look for la code again, just like the function above? Shouldn't it be trying to find the dataset as the function docstrings and title suggest?

File: liiatools_pipeline/sensors/location_schedule.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            hash_object.update(str(last_modified_time).encode())
+
+    return hash_object.hexdigest()
+
+
+def find_previous_matching_run(
+    run_records, run_key, la_path, dataset, key_op, key_folder
+):
+    previous_run_id = [
+        run.dagster_run.tags["dagster/run_key"]
```
Comment (cyramic) [inline liiatools_pipeline/sensors/location_schedule.py:81]: This is causing errors in my tests as the key doesn't exist when it is run using a schedule (which is how the live system will run)

Could you make sure the tag exists in a scheduled run as well?

**Review by cyramic — CHANGES_REQUESTED** [review]
Works really well the first run through, but the run_key tag is missing on scheduled runs causing issues when trying to run a second time.

**Comment by patrick-troy** [comment]
> This all seems to work just fine.
> 
> I have one question: In common.checks.check_dataset(), why does this seem to look for la code again, just like the function above? Shouldn't it be trying to find the dataset as the function docstrings and title suggest?

Aah yes this function is actually not even used so I will remove it. I first created it when we were going to check for the dataset in the folder name but we instead decided to loop through the allowed datasets which are defined in the .env file

**Comment by patrick-troy** [comment]
> Works really well the first run through, but the run_key tag is missing on scheduled runs causing issues when trying to run a second time.

Happy to add the error checking as you've suggested. I have been running this as a schedule though so not sure why it's not coming through. The only issue with your test because it can't find previous runs it will create a new run every time the schedule it triggered (even if there are no new files)

**Comment by cyramic** [comment]
It was merely the way I was (incorrectly) running it. Nothing to really worry about, but it did show a possible point where things could break. Better to catch potential issues & handle gracefully so the pipeline can continue

**Review by cyramic — APPROVED** [review]

**Pull request merged**